src/System.Net.Requests/src/System/Net/HttpDateParse.cs

```diff
@@ -398,5 +398,19 @@ public static bool ParseHttpDate(string dateString, out DateTime result)
 
             return true;
         }
+
+        // parse String to DateTime format.
+        public static DateTime StringToDate(String S)
+        {
+            DateTime dtOut;
+            if (HttpDateParse.ParseHttpDate(S, out dtOut))
+            {
+                return dtOut;
+            }
+            else
+            {
+                throw new ProtocolViolationException(SR.net_baddate);
+            }
+        }
```

Comment:
Nit: what you have is fine, but if you wanted it could also be consolidated to:
```C#
public static DateTime StringToDate(string s) =>
    HttpDateParse.ParseHttpDate(s, out DateTime dtOut) ? dtOut : throw new ProtocolViolationException(SR.net_baddate);
```

Reply:
Oh, I see, you were just moving it from another file.

Reply:
Yes, moving it but I'm gonna change it as I'm not the fan of verbosity